index.bs

```diff
@@ -1370,6 +1373,85 @@ object |transport|, run these steps.
 
 </div>
 
+# `WebTransportError` Interface #  {#web-transport-error-interface}
+
+<dfn interface>WebTransportError</dfn> is a subclass of {{DOMException}} that represents
+
+ - An error coming from the server or the network, or
+ - A reason for a client-initiated abort operation.
+
+<pre class="idl">
+[Exposed=(Window,Worker), SecureContext]
+interface WebTransportError : DOMException {
+  constructor(optional WebTransportErrorInit init = {});
+
+  readonly attribute octet? applicationProtocolCode;
+};
+
+dictionary WebTransportErrorInit {
+  octet applicationProtocolCode;
+  DOMString message;
+};
+</pre>
+
+## Internal slots  ## {#web-transport-error-internal-slots}
+
+A {{WebTransportError}} has the following internal slots.
+
+<table class="data" dfn-for="WebTransportError">
+ <thead>
+  <tr>
+   <th>Internal Slot
+   <th>Description (<em>non-normative</em>)
+  </tr>
+ </thead>
+ <tbody>
+  <tr>
+   <td><dfn>\[[ApplicationProtocolCode]]</dfn>
+   <td class="non-normative">The application protocol error code for this error, or null.
+  </tr>
+ </tbody>
+</table>
+
+## Constructor ##  {#web-transport-error-constructor1}
+
+<div algorithm>
+
+The <dfn constructor for="WebTransportError"
+lt="WebTransportError(init)">new WebTransportError(init)</dfn> constructor steps are:
+
+1. Let |error| be [=this=].
+1. Set |error|'s internal slots as:
+    : [=[[ApplicationProtocolCode]]=]
+    :: null
+1. Let |message| be `""`.
+1. Set |message| to |init|'s {{WebTransportErrorInit/message}} if it exists.
+1. [=WebTransportError/Set up=] |error| with |message|.
+1. Set |error|'s [=[[ApplicationProtocolCode]]=] to |init|'s
+   {{WebTransportErrorInit/applicationProtocolCode}} if it exists.
+
+</div>
+
+## Attributes ## {#web-transport-error-attributes}
+
+: <dfn for="WebTransportError" attribute>applicationProtocolCode</dfn>
+:: The getter steps are to return [=this=]'s [=[[ApplicationProtocolCode]]=].
```

Comment:
This seems a bit terse in describing what the new `error.applicationProtocolCode` is, or (for the constructor) how it should be used.

But I'm happy to iterate on that, since I suppose we still haven't fully decided (if it's the exact QUIC definition, or if we're going to reserve/overload some values (e.g. "other" instead of null)?